lib/jobs/copy-publickey.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+module.exports = copyKeyJobFactory;
+
+di.annotate(copyKeyJobFactory, new di.Provide('Job.CopyKey'));
+di.annotate(copyKeyJobFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Logger',
+    'Assert',
+    'Promise',
+    'Services.Waterline',
+    'ssh',
+    'JobUtils.Commands'
+));
+
+function copyKeyJobFactory(
+    BaseJob,
+    util,
+    Logger,
+    assert,
+    Promise,
+    waterline,
+    ssh,
+    CommandUtil
+) {
+    var logger = Logger.initialize(copyKeyJobFactory);
+    var commandUtil;
+    function CopyKeyJob(options, context, taskId) {
+        CopyKeyJob.super_.call(this, logger, options, context, taskId);
+        assert.string(this.context.target);
+        this.nodeId = this.context.target;
+        commandUtil = new CommandUtil(this.nodeId);
+    }
+    util.inherits(CopyKeyJob, BaseJob);
+
+    CopyKeyJob.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.needByIdentifier(self.nodeId)
+        .then(function(node) {
+            var commands = [
+                {cmd: 'mkdir -p .ssh'},
+                {cmd: 'echo '+node.sshSettings.publicKey+' >> .ssh/authorized_keys'}
+            ];
+            return Promise.map(commands, function(commandData) {
```

Comment:
Here it is applicable as you want to wait for the command utility to resolve before issuing the next command utility call.  Map will call the mapping function for each input item in whatever order they resolve and, unless you fiddle with concurrency, will not wait for the returned promise to resolve before calling the next iteration of the mapping function.

Mapseries and each will sequentially wait for the input promise to resolve and also wait for the mapping function's returned promise to resolve before iterating on the next item in the array.